data/rulesets.ts

```diff
@@ -964,7 +964,7 @@ export const Formats: {[k: string]: FormatData} = {
 							}
 							const forme = dex.getSpecies(formeName);
 							if (forme.changesFrom === originalForme.name && !forme.battleOnly) {
-								types = types.concat(forme.types);
+								types.push(...forme.types);
```

Comment:
This is a bug!!!!

Reply:
(`species.types` is not mutable)

Reply:
This would add these types after every validation.

So after three validations Meloetta would have types `Normal/Psychic/Normal/Fighting/Normal/Fighting/Normal/Fighting`.